src/SpeakDirectiveParser.js

```diff
@@ -1,17 +1,20 @@
 import IllegalArgumentError from "./errors/IllegalArgumentError";
 
 const httpMessageParser = require("http-message-parser");
-const { hasIn } = require("immutable");
+const { List, hasIn } = require("immutable");
 const util = require("util");
 
 /**
  * This method parses the multi-part AVS responses and extracts the
- * string representing Alexa's response. It strips away other information
+ * strings representing Alexa's responses. It strips away other information
  * like binary audio data, ssml tags etc.
  *
  * @param {String} alexaRawResponse The multi-part response from AVS. This
  * should not be empty or undefined.
  *
+ * @returns a List of text responses from Alexa. All the responses will be
+ * valid strings but a response can be empty if Alexa chooses to say nothing.
+ *
  * @throws IllegalArgumentError if Alexa's response couldn't be parsed out
  * of the input. The failure could be because the input is not well formatted
  * or doesn't contain the right fields to fetch Alexa's response.
```

Comment:
Add a TODO here on going through the AVS documentation with scrutiny and confirm that payload empty means an error.